neo4j/internal/bolt/bolt3.go

```diff
@@ -469,6 +472,9 @@ func (b *bolt3) Run(runCommand db.Command, txConfig db.TxConfig) (db.StreamHandl
 	if err := b.assertState(bolt3_streaming, bolt3_ready); err != nil {
 		return nil, err
 	}
+	if err := b.checkImpersonation(txConfig.ImpersonatedUser); err != nil {
```

Comment:
interesting, I probably would have left this out entirely :)

Reply:
I think there is a TestKit test for this case. We definitely agreed that the drivers should error if the user tries to impersonate and a Bolt 3 connection gets picked up from the pool.